src/composables/useViewer.js

```diff
@@ -80,10 +118,20 @@ export function useViewer() {
 		}
 	})
 
+	const generateViewerObject = (file) => ({
+		fileid: parseInt(file.id, 10),
+		filename: generateAbsolutePath(file.path),
+		basename: file.name,
+		mime: file.mimetype,
+		hasPreview: file.previewAvailable === 'yes' || file['preview-available'] === 'yes',
+		etag: file.etag,
+		permissions: generatePermissions(file.permissions),
+	})
```

Comment:
The fields should match the definition of fileInfo in Viewer
https://github.com/nextcloud/viewer/blob/af7635343546808dab410c76d57104723a7ca6cf/src/services/Viewer.js#L40
`etag` and `permissions` are extra ?

Reply:
Maybe, but without `etag` file isn't opening in Viewer, and permissions could be required for CRUD operations on the file.
Besides, the more information we provide to the Viewer, the lesser is chance for it to not fetch an additional data, or provide an image from cache. So I'd tend to leave the prop fields as-is